src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java

```diff
@@ -85,43 +87,48 @@ private Date getNextTime(Date lastActualExecutionTime) {
   @SneakyThrows
   private Date scheduleTaskWeekly(Date lastActualExecutionTime, Integer everyWeek) {
     ZonedDateTime startTime = ScheduleDateTimeUtil.convertScheduleTime(lastActualExecutionTime, scheduleParameters);
-    if (lastActualExecutionTime == null) {
-      everyWeek = 0;
+    startTime = findNextDayOfWeek(startTime);
+    if (Objects.nonNull(startTime) && everyWeek > 1) {
+      startTime = startTime.plusWeeks(everyWeek);
     }
-    startTime = findNextDayOfWeek(startTime, everyWeek);
     log.info("Weekly next schedule execution time in UTC for config {} is : {}", scheduleParameters.getId(), startTime);
     return ScheduleDateTimeUtil.convertToOldDateFormat(startTime, scheduleParameters);
   }
 
-  private ZonedDateTime findNextDayOfWeek(ZonedDateTime initZoneDateTime, Integer everyWeek) {
-    List<DayOfWeek> sortedDays = normalizeAndSortDayOfWeek();
-    if (!sortedDays.isEmpty() && sortedDays.get(0) != null) {
-      DayOfWeek firstDayOnTheWeek = sortedDays.get(0);
-      Iterator<DayOfWeek> dayOfWeekIterator = sortedDays.iterator();
-      DayOfWeek nextDayOfWeek = firstDayOnTheWeek;
-      while (dayOfWeekIterator.hasNext()) {
-        DayOfWeek dayOfWeek = dayOfWeekIterator.next();
-        if (dayOfWeek.getValue() > initZoneDateTime.getDayOfWeek().getValue()) {
-          nextDayOfWeek = dayOfWeek;
-          break;
+  private ZonedDateTime findNextDayOfWeek(ZonedDateTime scheduleDateTime) {
+    List<DayOfWeek> sortedChosenDays = normalizeAndSortDayOfWeek();
+
+    ZonedDateTime currentDateTime = getNowDateTime(scheduleParameters.getTimeZone());
+    DayOfWeek currentDay = currentDateTime.getDayOfWeek();
+
+    if (CollectionUtils.isNotEmpty(sortedChosenDays)) {
+      // iterate thought all days from Mon to Sun
+      for (DayOfWeek day: DayOfWeek.values()) {
+        // iterate thought chosen days
+        for (DayOfWeek chosenDay : sortedChosenDays) {
+          if (day == chosenDay) {
+            if (currentDay == chosenDay) {
+              if (currentDateTime.isBefore(scheduleDateTime)) {
+                // run scheduler in the same day
+                return scheduleDateTime;
+              } else {
+                return scheduleDateTime.plusDays(7); // schedule after a week
```

Comment:
possibly it's another issue: user selected Friday and Saturday 10am at 11am, job should be started on Saturday 10am, but will be started next Friday 10am

Reply:
thanks, writing unit test to cover this test case and possibly adjust business logic